### PR TITLE
Borg death fixes, human cleanup, carbon life tweaks

### LINKED ISSUE #2692: Borgs are Invincible
Simply put, borgs never die; they can keep moving around even if you drop their health to zero.

Not hard to understand why this is bad news.

### PR DESCRIPTION
**Pull request by tigercat2000**
fixes #2692 

The problem actually was that borgs called death() without returning
therefore the rest of Life() would run, and reset stat

More halloss implementations in human/Life() have been removed, halloss basically does nothing at all now, since it's unused.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -821,112 +821,67 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
...
-			silent = 0
-			return 1
-
-		// the analgesic effect wears off slowly
+		//the analgesic effect wears off slowly
 		analgesic = max(0, analgesic - 1)
 
-		//UNCONSCIOUS. NO-ONE IS HOME
-		if( (getOxyLoss() > 50) || (config.health_threshold_crit >= health) )
-			Paralyse(3)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/life.dm]: Guessing this and some other things above are handled elsewhere?

File: code/modules/mob/living/carbon/human/life.dm [hunk 2]
```diff
@@ -821,112 +821,67 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
...
 			else
 				hallucination -= 2
 
-
-		if(halloss > 100)
-			src << "<span class='notice'>You're in too much pain to keep going...</span>"
-			for(var/mob/O in oviewers(src, null))
-				O.show_message("<B>[src]</B> slumps to the ground, too weak to continue fighting.", 1)
-			Paralyse(10)
-			setHalLoss(99)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/life.dm]: the slow slow death of HalLoss

File: code/modules/mob/living/carbon/human/life.dm [hunk 3]
```diff
@@ -821,112 +821,67 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
...
 		if(paralysis)
 			blinded = 1
 			stat = UNCONSCIOUS
-			if(halloss > 0)
-				adjustHalLoss(-3)
+
 		else if(sleeping)
 			speech_problem_flag = 1
-			handle_dreams()
-			adjustStaminaLoss(-10)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/life.dm]: are these two procs handled elsewhere? They're both kinda important.

File: code/modules/mob/living/carbon/human/life.dm [hunk 4]
```diff
@@ -821,112 +821,67 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
...
-					sleeping = max(sleeping-1, 2)
-				else
-					sleeping = max(sleeping-1, 0)
+					sleeping = max(sleeping - 1, 2)
+
 			blinded = 1
 			stat = UNCONSCIOUS
-			if( prob(2) && health && !hal_crit )
-				spawn(0)
-					emote("snore")
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/life.dm]: Same here.

**Comment by Fox-McCloud** [comment]
Just a few concerns is all, making sure those are handled up the chain.

File: code/modules/mob/living/carbon/human/life.dm (hunk 1 above)
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/life.dm]: carbon/handle_regular_status_updates() specifically

File: code/modules/mob/living/carbon/human/life.dm (hunk 3 above)
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/life.dm]: yeah, they are called in carbon/handle_status_effects()

File: code/modules/mob/living/carbon/human/life.dm (hunk 4 above)
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/life.dm]: carbon/handle_status_effects() again

**Pull request merged**